lib/resolve-deps.js

```diff
@@ -30,47 +36,47 @@ var getDepsForFile = function (file, extraImports) {
   if (extraImports && extraImports.length > 0) {
     matches = extraImports.concat(matches);
   }
+
   if (matches.length > 0) {
-    _.each(matches, function (importedFile) {
-      var importedPath = path.resolve(path.dirname(file), importedFile);
+    let importedFile = await _.each(matches);
+    let importedPath = path.resolve(path.dirname(file), importedFile);
       // recursively get dependencies for imported files
-      deps[file].push(getDepsForFile(importedPath));
-    });
+    let importedDeps = await getDepsForFile(importedPath);
+    deps[file].push(importedDeps);
   }
   return deps;
-};
+}
+
+async function buildScriptFromDeps (deps) {
+
+  let script = '';
+  let [subDepsArray, file] = await _.each(deps);
+  if (subDepsArray.length > 0) {
+    let subDeps = await _.each(subDepsArray);
+    script += await buildScriptFromDeps(subDeps);
+  }
+  let readFile = await fs.readFile(file).toString('utf8');
```

Comment:
This needs to be in two stages... first await the `readFile` call, then translate to a `utf8` string. Otherwise you are going to stringify the Promise object, and end up with `[object Object]` as the output.